lib/controller/controller.go

```diff
@@ -149,7 +152,7 @@ const (
 	// DefaultResyncPeriod is used when option function ResyncPeriod is omitted
 	DefaultResyncPeriod = 15 * time.Minute
 	// DefaultThreadiness is used when option function Threadiness is omitted
-	DefaultThreadiness = 4
+	DefaultThreadiness = 1
```

Comment:
why?

Reply:
To ensure there is the only one operation in-flight for a given PVC.

Reply:
the work queue should take care of it. If you try to add the same PVC while it is in the queue nothing happens.

THe problem is not that more than one operation can be in-flight, it is that an operation can happen immediately after another under certain conditions. We can try to make controller smarter to avoid this (e.g. this is just a dumb example but wait x seconds after successful provision before removing PVC from queue) but technically it will be impossible to avoid entirely since update to a PVC can come from anywhere in the system and if it happens in the window between successful `Provision` and PV controller binding the PVC and PV, we will try to Provision again. Upstream pv controller has the same issue so whatever we want to do we should do it there first. https://github.com/kubernetes-incubator/external-storage/pull/937 should mitigate it.

Reply:
I've got it. I will close this pr.